CIP-XXXX/README.md

```diff
@@ -0,0 +1,447 @@
+update, their information.
+
+### Example 2: Token Metadata Insecurity
+
+One of the stated rationales for [CIP-68](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0068) was that the
+"original" Cardano NFT Metadata Standard ([CIP-25](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0025)) was
+insecure in some example use cases. This is due to the link between the transactional metadata and the minting of native
+assets. For example, a smart contract that cannot/does not validate against transaction metadata (i.e. Liquidity Pool
+tokens) could have a malicious user inject CIP-25 metadata into the transaction, potentially inserting illicit, illegal,
+or otherwise nefarious metadata information tied to the tokens which may be picked up and displayed to end users 
+unwittingly by explorers and wallets.
+
+### Example 3: De-duplication of Data
+
+Similar to [Example 1](#example-1--the-cardano-token-registry) above, when it comes to Non-Fungible Token (NFT) projects
+currently operating on the chain there is usually a desire to provide some level of information that pertains to all
+tokens under the given policy. Examples include project/collection names, social media handles, and miscellaneous
+project registration information. At current, this is generally solved by adding "static" fields in the metadata of
+every token. By moving this project-specific information a layer higher, not only do we achieve a path to dynamism
+but also reduce ledger bloat size by de-duplicating data. 
+
+Similarly, there are currently multiple marketplaces and decentralized exchanges (DEXes) in operation in the Cardano
+ecosystem. At current, most DEXes pull information from the _Cardano Token Registry_ but there is no similar function
+for NFT projects. As such, much information must be manually provided to individual marketplaces by the token projects
+creating an undue burden on the project creators to provide a largely static amount of information via different web
+forms and authentication schemas rather than simply publishing this information to the blockchain directly.
+
+## Specification
+
+### Registration Metadata Format
+
+A Native Script registration is a regular Cardano transaction with specific metadata attached (similar to CIP-36
+Catalyst Voter Registration). At the root level, the registration consists of two pieces of data:
+
+1. A Token Registration Payload Object [(TRPO)](#token-registration-payload-object)
+2. A Token Registration Witness array [(TRWA)](#token-registration-witness-array)
+
+### Token Registration Payload Object
+
+The Token Registration Payload Object (TRPO) consists of 4 required fields and then optional, additional fields to
+provide additional context and information as part of the registration payload. For the purposes of this draft CIP and
+in the examples below we will use the top-level JSON Metadata Index of **867** although this could be changed to the
```

Comment:
May I suggest to not use JSON for the specification, but actually use CBOR diagnostic since it better reflects what is stored on-chain. I'll make a concrete code suggestion for the relevant bits.

Reply:
Thanks for taking the time to discuss and help me see the benefits of describing the schema first in CDDL format to help better inform the JSON data structure so that we have a consistent data transport between both on-chain and JSON representation! Next push should include a first draft of the CDDL spec.